src/main/com/aca/homework/week9/calculator/SimpleCalculator.java

```diff
@@ -0,0 +1,40 @@
+package com.aca.homework.week9.calculator;
+
+public class SimpleCalculator {
+    private final StringSupplier supplier;
+
+    public SimpleCalculator(StringSupplier supplier) {
+        this.supplier = supplier;
+    }
+
+    public String askAndCalculate() {
+        System.out.println("Enter an expression in a format [digit][operation][digit]");
+        String expression = this.supplier.get();
+
+        if (!isValidExpression(expression)) {
+            System.out.println("error");
+            return "error";
+        }
+
+        int firstDigit = getDigitFromChar(expression.charAt(0));
+        int secondDigit = getDigitFromChar(expression.charAt(2));
+        char operation = expression.charAt(1);
+        return Operation.charToOperation(operation).performOperation(firstDigit, secondDigit);
+    }
+
+    public boolean isValidExpression(String expression) {
+        return expression.length() == 3 && isDigit(expression.charAt(0)) && isDigit(expression.charAt(2)) && isDigit(expression.charAt(1));
```

Comment:
expression.charAt(1) should be an operation, not a digit. I guess your tests are failing